packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -578,20 +471,7 @@ class DebuggerController extends DisposableController
   void _clearCaches() {
     _lastEvent = null;
     breakpointManager.clearCache();
-    _clearAutocompleteCaches();
-  }
-
-  void _clearAutocompleteCaches() {
-    _clazzCache.clear();
-    libraryMemberAutocompleteCache.clear();
-    libraryMemberAndImportsAutocompleteCache.clear();
-  }
-
-  /// Get the populated [Obj] object, given an [ObjRef].
-  ///
-  /// The return value can be one of [Obj] or [Sentinel].
-  Future<Obj> getObject(ObjRef objRef) {
-    return _service.getObject(_isolateRefId, objRef.id!);
+    evalService.cache.clear();
```

Comment:
this is badly coupled. Instead EvalService should listen for isolate changes itself instead of requiring the debuggerControl to call this method at the appropriate time.

Reply:
thanks. fixed.